### ISSUE TITLE
Allow plus addressing in Multi Inbox

### ISSUE THREAD

**Issue by BetaLyte** [AdhocAdam/smletsExchangeConnector#463]
**What's your idea? Go!**
Functionality of Multi Inboxes can be extended by allowing the use of plus addressing. Services and users can then use a plus address, instead of a separate inbox with redirect, to apply a template automatically at creation time.

**Additional context**
The regex in the MultipleMailboxes form just needs to accept an optional +whatever, ie. `^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*(\+[0-9a-zA-Z]([-.\w]*[0-9a-zA-Z]))*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})$
`

**Comment by AdhocAdam** [comment]
I had to read this a couple of times, but I think I'm following now. Initially it sounded like wholly new functionality for Multi-Mailbox, when in fact it's more of an enhancement as you're suggesting loosening the restriction on the textbox validation. In that you could use a "+" in an email address. Such as:

- mail@lab.lcl
- mail+@lab.lcl
- mail+incident@lab.lcl

Am I getting that right?

**Comment by BetaLyte** [comment]
Re-reading what I wrote, I can see that it's explained _just a bit_ convoluted.

Yes, I'm proposing changing the validation on the email field of Multi-Mailbox, so that it's possible to use plus-addressing, such as your examples (apart from nr. 2, that shouldn't be a valid email address).

We are already using it in our own env, and it's a lot easier, than creating additional mailboxes, just for the sake of redirecting the mails to apply templates.

**Comment by AdhocAdam** [comment]
I created a mailbox in Exchange and the second example above _is_ allowed. But given the purpose of the issue/PR you've raised, this is probably extraordinarily edge case. Just wanted to put this here for the sake of documentation.

![image](https://github.com/AdhocAdam/smletsexchangeconnector/assets/6636040/5881960d-5727-4129-9463-1d03ae023a19)

Otherwise, I tested this build out and looks like all is well for multi-mailbox configuration options. Current functionality is preserved and extended per your pull request. Second mailbox scenario is prevented within the Settings UI. 👍

![mailboxes02](https://github.com/AdhocAdam/smletsexchangeconnector/assets/6636040/99a7c42b-d3fb-4b33-9289-941c4e3ee2d8)

This will be included with the next release 😄
